examples/proc_GDS_pulse_capture.py

```diff
@@ -0,0 +1,171 @@
+            # slightly expand int range to include rising edges
+            int_range[0] -= 1e-6
+            int_range[-1] += 1e-6
+            beta["t_pulse", j] = (
+                abs(s["t":int_range]).integrate("t").data.item() * 1e6
+            )
+            # PR COMMENT: JF only read to here -- a bunch of stuff above were comments that weren't incorporated or obvious clean code stuff.  Please review from here to the end again
+            if skip_plots is not None and j % skip_plots == 0:
+                switch_to_plot(d, j)
+                fl.plot(
+                    abs(s["t":int_range]),
+                    color="black",
+                    label="integrated slice",
+                )
+                plt.ylabel(r"$\sqrt{P_{pulse}}$")
+                plt.text(
+                    int_range[0] * 1e6 - 1,
+                    -1,
+                    r"$t_{90} \sqrt{P_{tx}} = %f s \sqrt{W}$"
+                    % beta["t_pulse", j].item(),
+                )
+        # {{{ make nddata for beta of 90 pulse and 180 pulse
+        # {{{ beta vs t
+        fl.basename = None
+        fl.next(r"Measured $\beta$ vs A * $t_{pulse}$")
+        beta["t_pulse"] *= amplitude
+        beta.rename("t_pulse", "$A t_{pulse}$")
+        fl.plot(beta, "o", color=thiscolor, label=thislabel)
+        codehasbeenreviewed = False
+        if codehasbeenreviewed:
+            if amplitude > 1:
+                linear_threshold = 7
+                curve_threshold = 7.5
+            else:
+                linear_threshold = 2
+                curve_threshol = 2.5
+            if beta < linear_threshold:
+                mask = np.ones_like(beta.data, dtype=bool)
+            else:
+                mask = beta > curve_threshold
+            calibration_data = psd.nddata(
+                beta.data[mask], [-1], ["At_p"]
+            ).setaxis("At_p", beta.getaxis("At_p")[mask])
+            calibration_data.sort("At_p")
+            if beta < linear_threshold:
+                c = calibration_data.polyfit("At_p", order=10)
+            else:
+                c = calibration_data.polyfit("At_p", order=1)
+            fit_beta_v_t = np.polyval(c[::-1], beta.getaxis("At_p"))
+            fit = psd.nddata(fit_beta_v_t, "At_p").setaxis(
+                "At_p", beta.getaxis("At_p")
+            )
+            fl.plot(fit, color=thiscolor, ls=":", alpha=0.5)
+            psd.gridandtick(plt.gca())
+            plt.ylabel(r"measured $\beta$ / $\mathrm{\mu s \sqrt{W}}$")
+            plt.xlabel(r"programmed amplitude*$t_{pulse}$ / $\mu$s")
+            # }}}
+            # {{{ t vs beta
+            if amplitude > 1:
+                linear_threshold = 30
+                curve_threshold = 35
+            else:
+                linear_threshold = 10
+                curve_threshol = 11
+            t_v_beta = beta.shape.alloc().rename("At_p", "beta")
+            t_v_beta.setaxis("beta", beta.data)
+            t_v_beta.data[:] = beta.getaxis("At_p")
```

Comment:
this is the code to flip the function -- this should go up, before all the fitting stuff.  You want to fit functions for t_pulse(β).  Also feed `alloc(` `dtype=np.float64` as I do above (so that it's real and not complex)

Reply:
you should just be able to do something like  this:
```
c_nonlinear = t_v_beta.polyfit('t_pulse', order=10)
c_linear = t_v_beta['t_pulse':range_over_which_its_linear].polyfit('t_pulse',order=1)
```
that's it.

Reply:
As noted in the other PR, the `prog_plen` function is messed up in your PR.  I would not worry about that right now.   Just work it out here.

Reply:
Specifically, use this here:

```
def prog_plen(desired):
    def zonefit(desired):
        if desired > linear_threshold:
            return np.polyval(c_linear[::-1], desired)
        else:
            return np.polyval(c_nonlinear[::-1], desired)
    ret_val = np.vectorize(zonefit)(desired)
    if ret_val.size > 1:
        return ret_val
    else:
        return ret_val.item()
```
if you show here that it works (and show that the linear extension beyond the data works), then that basically gives you the code for the other PR!  You just need to hard-code in the linear_threshold and the coefficients for each power.

Reply:
I actually think that my comments above, and the lines that I'm highlighting, should form the basis of what you need to finish off this script.  I don't understand what all the rest of the lines of code in this `codehasbeenreviewed=False` block are for.

Reply:
fixed